src/oidcendpoint/oidc/userinfo.py

```diff
@@ -96,23 +94,26 @@ def do_response(self, response_args=None, request=None, client_id="", **kwargs):
         return {"response": resp, "http_headers": http_headers}
 
     def process_request(self, request=None, **kwargs):
-        _sdb = self.endpoint_context.sdb
-
+        _mngr = self.endpoint_context.session_manager
+        _session_info = _mngr.get_session_info_by_token(request["access_token"],
+                                                        user_session_info=True,
+                                                        client_session_info=True,
+                                                        grant=True)
+        token = _mngr.find_token(_session_info["session_id"], request["access_token"])
```

Comment:
We don't know if this really is an access token, it could also be a refresh token (I have tested this).

This probably should be changed in `parse_request` (https://github.com/IdentityPython/oidcendpoint/blob/27a0a4d27d9e53c4fb1e3bfd75dfb0a644bf7736/src/oidcendpoint/oidc/userinfo.py#L164)

Reply:
Added tests to verify right kind of token.